src/components/TestButtons/index.jsx

```diff
@@ -0,0 +1,99 @@
+      <div>User Info</div>
+      <br />
+      <Button onClick={() => UserInfoModel.create(['userid1', Templates.UserInfo])}>
+        Create user info
+      </Button>
+      <Button onClick={() => UserInfoModel.remove('userid1')}>Remove user info</Button>
+      <Button onClick={() => UserInfoModel.update({ displayName: 'Jan Doe' }, 'userid1')}>
+        Update user info
+      </Button>
+      <Button onClick={() => console.log(UserInfoModel.get('userid1'))}>Get user info</Button>
+      <div>User</div>
+      <br />
+      <Button onClick={() => UserModel.create(['userid1', Templates.User])}>Create user</Button>
+      <Button onClick={() => UserModel.update({ id: 'trolololol' }, 'userid1')}>Update user</Button>
+      <Button onClick={() => UserModel.remove('userid1')}>Remove user</Button>
+      <Button onClick={() => console.log(UserModel.get('userid1'))}>get user</Button>
+      <Button onClick={() => UserModel.addOwnedSnippetId('userid1', 'snippetid3')}>
+        add owned snippet id
+      </Button>
+      <Button onClick={() => UserModel.addEditableSnippetId('userid1', 'snippetid3')}>
+        aadd editable snippet id
+      </Button>
+      <Button onClick={() => UserModel.addViewableSnippetId('userid1', 'snippetid3')}>
+        add viewable snippet id
+      </Button>
+      <Button onClick={() => UserModel.removeOwnedSnippetId('userid1', 'snippetid3')}>
+        remove owned snippet id
+      </Button>
+      <Button onClick={() => UserModel.removeEditableSnippetId('userid1', 'snippetid3')}>
+        remove editable snippet id
+      </Button>
+      <Button onClick={() => UserModel.removeViewableSnippetId('userid1', 'snippetid3')}>
+        remove viewable snippet id
+      </Button>
+
+      <div>Snippets</div>
+      <br />
+      <Button onClick={() => SnippetModel.create(['snippetid3', Templates.Snippet])}>
+        create snippet
+      </Button>
+      <Button onClick={() => SnippetModel.update({ title: 'snippet lolol' }, 'snippetid3')}>
+        update snippet
+      </Button>
+      <Button onClick={() => SnippetModel.getByIDs(['snippetid3'])}>get snippets by id</Button>
+      <Button onClick={() => SnippetModel.remove('snippetid3')}>remove snippet</Button>
+      <Button onClick={() => SnippetModel.addTag('snippetid3', 'data structuresasdf')}>
+        add tags
+      </Button>
+      <Button onClick={() => SnippetModel.removeTag('snippetid3', 'data structuresasdf')}>
+        remove tag
+      </Button>
+      <Button onClick={() => SnippetModel.addCode('snippetid3', 'python', 'list.sort()')}>
+        add code
+      </Button>
+      <Button onClick={() => SnippetModel.addCode('snippetid3', 'cpp', 'cpcodepadsfadf ')}>
+        add code cpp
+      </Button>
+      <Button onClick={() => SnippetModel.removeCode('snippetid3', 'python')}>remove code</Button>
+      <Button onClick={() => SnippetModel.removeCode('snippetid3', 'cpp')}>remove code cpp</Button>
+    </Grid>
+  );
```

Comment:
I see there are repetitive modules accessing methods. e.g `SnippetModel.create`, `SnippetModel.update` and so on. 

Usually if you end up repeating the reference of an object what we would do to avoid that is by destructuring it like. `const { create, update, remove } = SnippetModel`. But doing it this way with the current code, would pose a lot of name conflicts because all of the models have the same method names ( create, update, remove ). So it would be good if we name the methods unique to what they are actually doing. 

So we might end up with something like

```js
import { createSnippetAccess, updateSnippetAccess, removeSnippetAccess } from './SnippetAccessModel'
import { createSnippet, updateSnippet, removeSnippet } from './SnippetModel'
import { createUser, updateUser, removeUser } from './UserModel'
```

Reply:
`export default {
  createSnippet,
  updateSnippet,
  getSnippet,
  getSnippetsByIds,
  getAllSnippets,
  removeSnippet,
  addSnippetTag,
  removeSnippetTag,
  addSnippetCode,
  removeSnippetCode,
};`
I updated the fn names to be more explicit.

Reply:
A few more things, don't inline the button click functions, make a function for them then pass those in the button

```js
function createSnippet(){
 // firebase add snippet code
}

<Button onClick={createSnippet}>Add snippet</Button>
```

Reply:
Let's destructure the methods from the models? So you don't have to do 
```js
SnippetModel.create
SnippetModel.edit
SnippetModel.remove
```
and have something like this instead
```js
import {createSnippet, removeSnippet, editSnippet} from 'snippetModel'
or
import SnippetModel from 'snippetModel'
const {createSnippet, removeSnippet, editSnippet} = SnippetModal
```
I know that this is a test file, but ideally if you only need the methods inside the object and not the main object i.e `SnippetModel` let's just desctructure the contents that we need to avoid repetition.